Added ability to register commands using simple strings (fixes #2806)

6A opened the pull request:
**What this PR does / why we need it**:
Enables the syntax specified in #2806 to work.

**Which issue(s) this PR fixes**
#2806.

**Notes:**
This simply makes the "command" object in `IKeyRemapping` be either the old object or a string. If it's a string, we then interpret the string as `{ command: str; args: [] }`.

xconverge commented:
This looks good to me, I like the simplicity to the user

I think a note in the readme as an example would be nice

jpoon commented on README.md:
```diff
@@ -245,16 +245,36 @@ Custom remappings are defined on a per-mode basis.
-        ]
-    }
-]
+    "vim.normalModeKeyBindingsNonRecursive": [
```
Awesome! Would you mind updating the other examples to use the more simplified version?

I'd also be sure to leave examples of how to specify multiple commands and such with both notations.

6A replied:
Do you have an example of a command which also take arguments? I tried finding one to show a "complete" object but didn't succeed. 

In may case I'll change the README in a way that suggests the string is the default, and then when the command takes additional arguments you can use the fully fledged object.

6A commented:
Alright, I changed all the examples in the README to use that syntax, and also added a command that uses arguments to show the difference (`git.clone` with this repository).

xconverge commented:
Thanks!
